### ISSUE TITLE
Doesn't seem to work with personal access token grant type

### ISSUE THREAD

**Issue by rbreahna** [sfelix-martins/passport-multiauth#32]
I've implemented this package but it seems to not work with personal access tokens. Tokens are being stored with the users provider regardless of the guard applied

**Comment by sfelix-martins** [comment]
@rbreahna I will check this. Thanks

**Comment by sfelix-martins** [comment]
@rbreahna I tested and I can use the package with personal access tokens. Follow a implementation sample:

```php
Route::middleware('oauth.providers')->post('/token', function (Request $request) {    
    $model = \App\User::class;
    if ($request->provider === 'admins') {
        $model = \App\Admin::class;
    }

    // You should validade the user credentials. 
    // For testing purposes I just will validate if email exists.
    $user = $model::where('email', $request->email)->first();
    if (! $user) {
        throw new \Illuminate\Auth\AuthenticationException();
    }

    $accessToken = $user->createToken('My Token')->accessToken;

    return [
        'accessToken' => $accessToken,
    ];
});

Route::middleware('auth:api')->get('/user', function (Request $request) {
    return get_class($request->user());
});

Route::middleware('auth:admin')->get('/admin', function (Request $request) {
    return get_class($request->user());
});
```

And the requests sample:

- Create token passing param `provider`. This parameter is required to your route using `oauth.providers` middleware. This package middleware set the correct provider to access token that will be created:

```http
POST /api/token HTTP/1.1
Host: homestead.test
Accept: application/json
Content-Type: application/x-www-form-urlencoded
Cache-Control: no-cache

{
    "email" : "smartins@mail.com",
    "provider" : "admins"
}
```

Response:
```json
{
    "accessToken": "accessTokenCreated",
}
```

- Login on route with `auth` middleware passing the `admin` guard using the access token created on route above:

```http
GET /api/admin HTTP/1.1
Host: homestead.test
Authorization: Bearer accessTokenCreated
Accept: application/json
Cache-Control: no-cache
```

Response: 200 OK "App\Admin"

- Login on route with `auth` middleware passing the `api` guard:

```http
GET /api/user HTTP/1.1
Host: homestead.test
Authorization: Bearer accessTokenCreated
Accept: application/json
Cache-Control: no-cache
```

Response: 401
```json
{
    "message": "Unauthenticated."
}
```

I hope it helps. Feels free to ask.

**Comment by apoorvm8** [comment]
I tried the above code, but it doesn't seem to work, for some reason $request->provider is not working for me.

**Comment by sfelix-martins** [comment]
Make sure that you are passing he param `provider` on request.

**Comment by apoorvm8** [comment]
Sorry, I am a beginner in Laravel, can you give the proper code. Like showing how you are passing the param provider on request.
Thank for your help and time

Edit: Ah, It worked I wasn't passing the param. Thank you for your time.

**Comment by sfelix-martins** [comment]
Thank you @apoorvm8. I'm available. Closes #38 #43
